example/gin/main.go

```diff
@@ -0,0 +1,52 @@
+// (c) Copyright IBM Corp. 2021
+// (c) Copyright Instana Inc. 2016
+
+// +build go1.11
+
+package main
+
+import (
+	"flag"
+
+	"github.com/gin-gonic/gin"
+	instana "github.com/instana/go-sensor"
+	"github.com/instana/go-sensor/instrumentation/instagin"
+)
+
+const (
+	defaultPort    = "8881"
+	defaultAddress = "localhost"
+)
+
+func main() {
+	port := flag.String("port", defaultPort, "port to use by an example")
+	address := flag.String("address", defaultAddress, "address to use by an example")
+
+	flag.Parse()
+	engine := gin.Default()
+
+	// create a sensor
+	sensor := instana.NewSensor("gin-sensor")
+
+	// add middleware to the gin handlers
+	instagin.AddMiddleware(sensor, engine)
+
+	engine.GET("/myendpoint", func(c *gin.Context) {
+		c.JSON(200, gin.H{
+			"message": "pong",
+		})
+	})
+
+	// use group: v1
+	v1 := engine.Group("/v1")
+
+	{
```

Comment:
Does Gin require to use a separate block to set up the handlers? Otherwise I'd suggest we omit it to simplify the example.

Reply:
It does not. I will change it in child PR.